course-project/Backend/src/cmd/rest-api/lib/proxy_test.go

```diff
@@ -0,0 +1,88 @@
+package lib
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestGetCurrentWeatherFromAPI(t *testing.T) {
+	type args struct {
+		cityName string
+	}
+	tests := []struct {
+		name string
+		args args
+		want Weather
+	}{
+		{name: "london", args: args{cityName: "london"}, want: Weather{
+			CityName:    "London",
+			Temperature: 0.0,
+		}},
+		{name: "astana", args: args{cityName: "astana"}, want: Weather{
+			CityName:    "Nur-Sultan",
+			Temperature: 0.0,
+		}},
+		{name: "almaty", args: args{cityName: "almaty"}, want: Weather{
+			CityName:    "Almaty",
+			Temperature: 0.0,
+		}},
+		{name: "asdasda", args: args{cityName: "asdasda"}, want: Weather{
+			CityName:    "",
+			Temperature: 0.0,
+		}},
+		{name: "empty", args: args{cityName: "NULL"}, want: Weather{
```

Comment:
You changed the empty string to "NULL" for a case. Still, what about the test case for an empty string? Should probably test for that as well.